jina/jaml/parsers/executor/legacy.py

```diff
@@ -78,7 +78,8 @@ def parse(self, cls: Type['BaseExecutor'], data: Dict) -> 'BaseExecutor':
         arguments_from_cls = LegacyParser._get_all_arguments(cls)
         arguments_from_yaml = set(data.get('with', {}))
         difference_set = arguments_from_yaml - arguments_from_cls
-        if any(difference_set):
+        # only log warnings about unknown args for main Pea
+        if any(difference_set) and LegacyParser.is_tail_or_head(data):
```

Comment:
I somehow dont get this one? Your comment says you only want to do this for a main pea but then you enter the if here if it is tail/head? I guess I am missing something here?

Reply:
I fixed that. Initially I wanted the method to return if it's a main Pea. But switched it around and forgot to change here. The pains of not having a proper test for logging